pkg/environ/providers/vault/main.go

```diff
@@ -53,22 +56,15 @@ func New() (environ.Provider, error) {
 
 	log.Debugf("Creating vault api client. addr=%v", os.Getenv("VAULT_ADDR"))
 
-	clientTimeout, err := time.ParseDuration(os.Getenv("VAULT_CLIENT_TIMEOUT"))
-
-	if err != nil {
-		clientTimeout = time.Second * 3
+	vcc := api.DefaultConfig()
+	if t := os.Getenv(api.EnvVaultClientTimeout); t == "" {
+		vcc.Timeout = defaultClientTimeout
 	}
-
-	clientMaxRetries, err := strconv.Atoi(os.Getenv("VAULT_MAX_RETRIES"))
-
-	if err != nil {
-		clientMaxRetries = 1
+	if r := os.Getenv(api.EnvVaultMaxRetries); r == "" {
+		vcc.MaxRetries = defaultClientRetries
 	}
 
-	vaultConfig := api.DefaultConfig()
-	vaultConfig.HttpClient.Timeout = clientTimeout
-	vaultConfig.MaxRetries = clientMaxRetries
-	vc, er := api.NewClient(vaultConfig)
+	vc, er := api.NewClient(vcc)
```

Comment:
Refactoring how we are setting the default vault client values here. `api.DefaultConfig` will pull the values out of the `VAULT_*` env vars already, so just set the values if the env vars are not set.